card/src/main/java/com/adyen/checkout/card/internal/ui/view/StoredCardView.kt

```diff
@@ -0,0 +1,173 @@
+    }
+
+    private fun initLocalizedStrings(localizedContext: Context) {
+        binding.textInputLayoutCardNumber.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Card_CardNumberInput,
+            localizedContext
+        )
+        binding.textInputLayoutExpiryDate.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Card_ExpiryDateInput,
+            localizedContext
+        )
+        binding.textInputLayoutSecurityCode.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Card_SecurityCodeInput,
+            localizedContext
+        )
+    }
+
+    private fun observeDelegate(delegate: CardDelegate, coroutineScope: CoroutineScope) {
+        delegate.outputDataFlow
+            .onEach { outputDataChanged(it) }
+            .launchIn(coroutineScope)
+    }
+
+    private fun initSecurityCodeInput() {
+        val securityCodeEditText = binding.textInputLayoutSecurityCode.editText as? SecurityCodeInput
+        securityCodeEditText?.setOnChangeListener { editable: Editable ->
+            cardDelegate.updateInputData { securityCode = editable.toString() }
+            binding.textInputLayoutSecurityCode.hideError()
+        }
+        securityCodeEditText?.onFocusChangeListener = OnFocusChangeListener { _, hasFocus ->
+            val securityCodeValidation = cardDelegate.outputData.securityCodeState.validation
+            if (hasFocus) {
+                binding.textInputLayoutSecurityCode.hideError()
+            } else if (securityCodeValidation is Validation.Invalid) {
+                binding.textInputLayoutSecurityCode.showError(localizedContext.getString(securityCodeValidation.reason))
+            }
+        }
+    }
+
+    private fun outputDataChanged(cardOutputData: CardOutputData) {
+        binding.editTextCardNumber.setText(
+            localizedContext.getString(
+                R.string.card_number_4digit,
+                cardOutputData.cardNumberState.value
+            )
+        )
+        binding.editTextExpiryDate.date = cardOutputData.expiryDateState.value
+        binding.textInputLayoutSecurityCode.takeIf { isVisible }?.requestFocus()
+
+        setDetectedCardBrand(cardOutputData)
+    }
+
+    private fun setDetectedCardBrand(cardOutputData: CardOutputData) {
+        val detectedCardTypes = cardOutputData.detectedCardTypes
+        binding.cardBrandLogoImageViewPrimary.strokeWidth = RoundCornerImageView.DEFAULT_STROKE_WIDTH
+        binding.cardBrandLogoImageViewPrimary.loadLogo(
+            environment = cardDelegate.componentParams.environment,
+            txVariant = detectedCardTypes[0].cardBrand.txVariant,
+            placeholder = R.drawable.ic_card,
+            errorFallback = R.drawable.ic_card,
+        )
+    }
+
+    override fun highlightValidationErrors() {
+        cardDelegate.outputData.let {
+            val securityCodeValidation = it.securityCodeState.validation
+            if (securityCodeValidation is Validation.Invalid) {
```

Comment:
We also have to show the actual error with `binding.textInputLayoutSecurityCode.showError(...)`

Reply:
nice catch! I will update here.